web/src/settings_org.ts

```diff
@@ -777,25 +804,189 @@ export function deactivate_organization(e: JQuery.Event): void {
     e.preventDefault();
     e.stopPropagation();
 
+    let custom_deletion_time_input = realm.server_min_deactivated_realm_deletion_days;
+    let custom_deletion_time_unit = "days";
     function do_deactivate_realm(): void {
+        const raw_delete_in = $<HTMLSelectOneElement>("select:not([multiple])#deletes_in").val()!;
+        let deletes_in_days: number | null;
+
+        // See settings_config.realm_deletion_in_values for why we do this conversion.
+        if (raw_delete_in === "null") {
+            deletes_in_days = null;
+        } else if (raw_delete_in === "custom") {
+            const deletes_in_minutes = settings_ui.get_expiration_time_in_minutes(
+                custom_deletion_time_unit,
+                custom_deletion_time_input,
+            );
+            deletes_in_days = deletes_in_minutes / (60 * 24);
+        } else {
+            const deletes_in_minutes = Number.parseFloat(raw_delete_in);
+            deletes_in_days = deletes_in_minutes / (60 * 24);
+        }
+        const data = {
+            deletion_delay_days: JSON.stringify(deletes_in_days),
+        };
+
         channel.post({
             url: "/json/realm/deactivate",
+            data,
             error(xhr) {
                 ui_report.error($t_html({defaultMessage: "Failed"}), xhr, $("#dialog_error"));
             },
         });
     }
 
-    const html_body = render_settings_deactivate_realm_modal();
+    const time_unit_choices = ["days", "weeks"];
+    const minimum_allowed_days = realm.server_min_deactivated_realm_deletion_days;
+    const maximum_allowed_days = realm.server_max_deactivated_realm_deletion_days;
+
+    const delete_in_options: DeleteOptions[] = [];
```

Comment:
Since `settings_org` was recently converted from js to ts, I am not really sure if this is the correct way to filter only the vaid `delete in options` into the dropdown.

<details><summary>Delete in options:</summary>
<p>

![image](https://github.com/user-attachments/assets/f6881b28-8b76-459c-9e2f-e26f475738d2)


</p>
</details>